### ISSUE TITLE
Async 'hello world' example doesn't work

### ISSUE THREAD

**Issue by artemkin** [mirage/ocaml-cohttp#324]
https://github.com/mirage/ocaml-cohttp/blob/master/examples/async/hello_world.ml#L13

`Uri.path uri` returns `/test%3Fhello=xyz`, so it is never matched.

```
match Uri.path uri with
  | "/test" ->
       Uri.get_query_param uri "hello"
    |> Option.map ~f:(fun v -> "hello: " ^ v)
    |> Option.value ~default:"No param hello supplied"
    |> Server.respond_with_string
  | _ ->
    Server.respond_with_string ~code:`Not_found "Route not found"
```

**Comment by dsheets** [comment]
This has been fixed by #318 which is awaiting merge.

**Comment by dsheets** [comment]
I should say that only 0.16.1 is broken in this way. 0.16.0 is fine and 0.17.0 (to be released later today) will have this issue corrected. Thanks for your report!
